google-cloud-pubsub/lib/google/cloud/pubsub/async_publisher.rb

```diff
@@ -360,9 +355,9 @@ def publish_batch_sync topic_name, batch
             # Cancel the batch if the error is not to be retried.
             begin
               raise OrderingKeyError, batch.ordering_key
-            rescue OrderingKeyError => ordered_stopped
+            rescue OrderingKeyError => e
               # Set the error to OrderingKeyError
-              e = ordered_stopped
+              e = e
```

Comment:
Is the `e` variable is shadowing something further up?

Reply:
Better declare `OrderingKeyError => e` as `OrderingKeyError => e2` or `OrderingKeyError => oe`

Reply:
This is one of those things Rubocop changed that I didn't check. It looks like it will shadow and update the `e` from before the block.

```ruby
begin
1 / 0
rescue ZeroDivisionError => e
  begin
    "".freeze.upcase!
  rescue FrozenError => e
  end
  puts e.class
end
# => FrozenError
```

I don't like it, but I don't dislike it enough to add an exception for it.